### ISSUE TITLE
iCloud Private Relay Service doesn't respond with NXDomain properly

### ISSUE THREAD

**Issue by devipasigner** [AdguardTeam/AdGuardHome#5286]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not ask a question

### Operating system type

Linux, Other (please mention the version in the description)

### CPU architecture

AMD64

### Installation

GitHub releases or script from README

### Setup

On one machine

### AdGuard Home version

108 b23

#### What did you do?
Enable the sevice 'icloud private relay' in Filters -> Blocked services
#### Expected result
Response with nxdomain in query logs and dig
#### Actual result
Often times get 0.0.0.0 or NOERROR

#### Additional information
 Doesn't happen if manually put the rules in custom filtering rules with $important flag specified

**Comment by ainar-g** [comment]
Hello.  We cannot reproduce this.  Are you sure that there isn't a rule that is triggered instead of the services one?  What rule do you see in the query log?

**Comment by devipasigner** [comment]
> Hello. We cannot reproduce this. Are you sure that there isn't a rule that is triggered instead of the services one? What rule do you see in the query log?

Nope, I do have a blocklist specifly for DoH domains but its not affecting anything as you can see in the query log outputs.
![image](https://user-images.githubusercontent.com/113001348/209240370-bffc2155-d791-4c58-a689-7d135b591576.png)

**Comment by devipasigner** [comment]
If I run dig it also shows the same. 0.0.0.0 and not NXDomain.
![image](https://user-images.githubusercontent.com/113001348/209589438-16e44ffe-eafa-49b1-9b6a-0796f450febb.png)

**Comment by devipasigner** [comment]
If I manually add to my custom filtering rules:
! Disable iCloud Private Relay
||mask.icloud.com^$important,dnsrewrite=NXDOMAIN;;
||mask-h2.icloud.com^$important,dnsrewrite=NXDOMAIN;;
![image](https://user-images.githubusercontent.com/113001348/209589552-e37d88fd-e256-4560-9d23-a3b2f84c33e3.png)

**Comment by jballer** [comment]
+1. I just ran into this same scenario: the rewrite is not processed when it comes from the blocked services JSON, but the same rules correctly rewrite when pasted in as custom filter rules.

**Comment by jballer** [comment]
For more context on why this is a problem: the current behavior breaks client UX by leaving requests hanging until a lengthy timeout occurs.

With `NXDOMAIN` the client _immediately_ recognizes that private relay is not available.

On iOS, this results in a "private relay unavailable" alert. On macOS, there's a confirmation step that asks whether you want to continue using the network.

**Comment by devipasigner** [comment]
Yes, I believe that maybe the services are not able to process the dns rewrite rules.

Should be fixed!

**Comment by devipasigner** [comment]
Ping @ainar-g for update..

**Comment by bayasdev** [comment]
> If I manually add to my custom filtering rules:
> ! Disable iCloud Private Relay
> ||mask.icloud.com^$important,dnsrewrite=NXDOMAIN;;
> ||mask-h2.icloud.com^$important,dnsrewrite=NXDOMAIN;;
> ![image](https://user-images.githubusercontent.com/113001348/209589552-e37d88fd-e256-4560-9d23-a3b2f84c33e3.png)
> 

+1 this worked for me

**Comment by levid0s** [comment]
Having the same issue.
What data is this issue waiting for?

```
$ nslookup
> mask.icloud.com
Server:         192.168.1.1
Address:        192.168.1.1#53

Non-authoritative answer:
Name:   mask.icloud.com
Address: 0.0.0.0
```

```
14:20:13   mask.icloud.com   Blocked service   192.168.1.11
```

**Comment by drrlvn** [comment]
Just encountered this myself and switched to custom rules as a workaround. Would appreciate a fix if possible.
